### PR TITLE
add a sample project about how to use targeting filter

### PR DESCRIPTION
**Pull request by ivywei0125**
## Purpose
Add a sample project about how to use targeting filter

### CODE REVIEW

File: appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-targeting-filter-web-sample/src/main/java/com/example/demo/configuration/AppConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import com.example.demo.impl.TargetingContextImpl;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+@Configuration
+public class AppConfiguration {
+
+    @Bean(name = "targetingFilter")
+    @Scope("request")
```
Comment (mrm9084) [inline appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-targeting-filter-web-sample/src/main/java/com/example/demo/configuration/AppConfiguration.java:13]: NIT: You don't need the name value if the field matches method name.
NIT: You can use `@RequestScope` instead.

File: appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-targeting-filter-web-sample/src/main/java/com/example/demo/controllers/HelloController.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+
+import com.azure.spring.cloud.feature.management.FeatureManager;
+import org.springframework.web.bind.annotation.GetMapping;
+
+
+@Controller
+@ConfigurationProperties("controller")
```
Comment (mrm9084) [inline appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-targeting-filter-web-sample/src/main/java/com/example/demo/controllers/HelloController.java:12]: You shouldn't need this line it isn't an `@ConfigurationProperties`

File: appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-feature-management-web-sample/src/main/resources/application.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ spring:
           delay: 5s
```
Comment (mrm9084) [inline appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-feature-management-web-sample/src/main/resources/application.yml:9]: Can you add a readme explaining how to run and what is happening?

**Review by mrm9084 — CHANGES_REQUESTED** [review]

File: appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-targeting-filter-web-sample/src/main/java/com/example/demo/configuration/AppConfiguration.java (hunk 1 above)
Comment (ivywei0125) [reply appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-targeting-filter-web-sample/src/main/java/com/example/demo/configuration/AppConfiguration.java]: Updated

File: appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-targeting-filter-web-sample/src/main/java/com/example/demo/controllers/HelloController.java (hunk 2 above)
Comment (ivywei0125) [reply appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-targeting-filter-web-sample/src/main/java/com/example/demo/controllers/HelloController.java]: Updated

File: appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-feature-management-web-sample/src/main/resources/application.yml (hunk 3 above)
Comment (ivywei0125) [reply appconfiguration/spring-cloud-azure-feature-management-web/spring-cloud-azure-feature-management-web-sample/src/main/resources/application.yml:9]: Sure. Added README.md

**Review by mrm9084 — APPROVED** [review]

**Pull request merged**